apm-agent-plugins/apm-jms-plugin/apm-jms-plugin-base/src/main/java/co/elastic/apm/agent/jms/JmsMessageListenerInstrumentation.java

```diff
@@ -46,6 +53,28 @@ public class JmsMessageListenerInstrumentation extends BaseJmsInstrumentation {
     @SuppressWarnings("WeakerAccess")
     public static final Logger logger = LoggerFactory.getLogger(JmsMessageListenerInstrumentation.class);
 
+    @Nullable
+    private JmsConfiguration configuration;
+
+    @Override
+    public ElementMatcher<? super NamedElement> getTypeMatcherPreFilter() {
+        if (configuration == null) {
+            configuration = GlobalTracer.getTracerImpl().getConfig(JmsConfiguration.class);
+        }
```

Comment:
[minor] Instrumentations can declare a constructor that receives a `co.elastic.apm.agent.impl.ElasticApmTracer`, consider using that instead of such lazy initialization.